pkg/generator/schema_generator.go

```diff
@@ -373,7 +415,7 @@ func (g *schemaGenerator) generateType(
 		return codegen.EmptyInterfaceType{}, nil
```

Comment:
Thanks for maintaining this great tool!
I have been trying to use it for a schema with `anyOf` and I have a comment on this PR. 

When using `anyOf`, the `type` may not be set, and only be defined in the items which are part of the `anyOf` itself - see for instance https://json-schema.org/understanding-json-schema/reference/combining#anyOf.

Specifically, I tried running `go-jsonschema` from the lastest release as well as from this PR for this [schema](https://github.com/cdevents/spec/blob/main/schemas/links/embeddedlinksarray.json), and in both cases, I hit the `EmptyInterfaceType`.

```golang
type EmbeddedlinksarrayJson []interface{}
```

Reply:
@afrittoli so without a hinted type the library would have to use some heuristics to determine the type where possible. If that information is not available, or if the types are not compatible, the empty interface is all it can use to satisfy the type constraints. I could add that to make things a bit better, but it will only cover a handful of cases.